packages/synapse-react-client/src/components/SynapseTable/SynapseTableUtils.ts

```diff
@@ -114,3 +115,22 @@ export function isSortableColumn(column: ColumnType) {
       return false
   }
 }
+
+export function getDefaultPrimaryKey(
+  entity?: Table,
+  columnModels?: ColumnModel[],
+): string[] | undefined {
+  if (
+    isFileViewOrDataset(entity) &&
+    columnModels?.find(cm => cm.name === 'id')
```

Comment:
Also verify that the id column model type is entity id?

Reply:
Since it's a view I think the data in this column has to be the unique synID, so it shouldn't matter if it's entity ID or string (or number, if that cast works?) because this is used for row selection.